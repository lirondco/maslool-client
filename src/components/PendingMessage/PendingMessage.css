@media screen and (min-width: 500px) {
  .PendingMessage {
    margin-bottom: 2%;
    padding: 1%;
    border: #524c57 solid;
    border-radius: 10px;
    box-shadow: 5px 5px 2.5px grey;
    display: flex;
    flex-direction: row;
    background-color: #fffcf5;
  }
  .PendingMessage button {
    border-radius: 10px;
    margin-left: 0 !important;
    margin-top: 0 !important;
    margin-bottom: 0 !important;
    font-weight: bold;
    background-color: #f3f4f5;
    margin-right: 1% !important;
  }

  .PendingMessage button:hover {
    border-radius: 10px;
    margin-left: 0 !important;
    margin-top: 0 !important;
    margin-bottom: 0 !important;
    font-weight: bold;
    background-color: red;
    color: white;
    margin-right: 1% !important;
  }

  .outer_item {
    background-color: #fae5cf;
    border-radius: 10px;
    border: slategray solid 2px;
    padding: 1%;
    width: 20%;
    font-weight: bold;
    text-align: right;
    overflow-wrap: break-word;
  }
  .outer_item span {
    background-color: #fae5cf;
    font-style: italic;
    text-decoration: underline;
  }
  .outer_item a {
    background-color: #fae5cf;
    text-decoration: underline;
    color: blue;
  }
  .outer_item a:hover {
    color: purple;
  }

  .inner_item {
    border-radius: 10px;
    border: slategray solid 2px;
    padding: 1%;
    width: 600px !important;
    font-weight: bold;
    margin: 0 1% 0 1%;
    text-align: left;
  }
  .inner_item a span {
    border-radius: 10px;
    background-color: white;
    padding: 1%;
    border: gray solid 1px;
    text-decoration: underline;
    color: maroon;
  }
  .inner_item a span:hover {
    color: darkblue;
  }
}

@media screen and (max-width: 499.999px) {
  .PendingMessage {
    margin-bottom: 2%;
    padding: 1%;
    border: #524c57 solid;
    border-radius: 10px;
    box-shadow: 5px 5px 2.5px grey;
    display: flex;
    flex-direction: column;
    background-color: #fffcf5;
  }
  .PendingMessage button {
    margin-left: 0 !important;
    margin-top: 0 !important;
    margin-bottom: 0 !important;
    border-radius: 50%;
    font-weight: bold;
    width: fit-content;
    background-color: #f3f4f5;
    margin-right: 1% !important;
  }
  .outer_item {
    background-color: #fae5cf;
    border-radius: 10px;
    border: slategray solid 2px;
    padding: 1%;
    font-weight: bold;
  }
  .outer_item span {
    background-color: #fae5cf;
    font-style: italic;
    text-decoration: underline;
  }
  .outer_item a {
    background-color: #fae5cf;
    text-decoration: underline;
    color: blue;
  }
  .outer_item a:hover {
    color: purple;
  }
  .inner_item {
    border-radius: 10px;
    border: slategray solid 2px;
    padding: 1%;
    width: 90%;
    font-weight: bold;
    margin: 3%;
    display: flex;
    text-align: center;
    flex-direction: column;
  }
  .inner_item a span {
    border-radius: 10px;
    padding: 1%;
    text-decoration: underline;
    color: maroon;
  }
  .inner_item a span:hover {
    color: darkblue;
  }
}
