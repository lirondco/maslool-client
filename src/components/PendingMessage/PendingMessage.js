import React, { Component } from "react";
import { Link } from "react-router-dom";
import "./PendingMessage.css";

export default class PendingMessage extends Component {
    static defaultProps = {
        onDeleteClick: () => {},
    };

    handleDeleteClick = (ev) => {
        ev.preventDefault();
        this.props.onDeleteClick();
    };

    render() {
        const pending = this.props.pending;
        return ( <
            div className = "PendingMessage" >
            <
            button aria - label = "delete"
            onClick = { this.handleDeleteClick } >
            x <
            /button> <
            p className = "outer_item" >
            From: < span > { pending.user.username } < /span> <
            /p> <
            p className = "inner_item" >
            Content: { " " } <
            Link to = { `/messages/${pending.id}` } >
            <
            span > { pending.message.substring(0, 20) }...click to see more <
            /span> <
            /Link> <
            /p> <
            p className = "outer_item" >
            Contact: { " " } <
            span >
            <
            a href = { `mailto:${pending.user.email}` } > { pending.user.email } < /a> <
            /span> <
            /p> <
            /div>
        );
    }
}