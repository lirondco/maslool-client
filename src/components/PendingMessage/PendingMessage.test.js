import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import PendingMessage from "./PendingMessage";

describe("PendingMessage component", () => {
  const props = {
    id: 9,
    message: "This is a message",
    user: {
      id: 1,
      username: "test_user",
      email: "test@user.no",
      join_date: "2021-03-11T06:47:06.477Z",
      admin: false,
      banned: false,
    },
  };

  it("renders the message given props", () => {
    const wrapper = shallow(<PendingMessage pending={props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('button works when clicked', () => {
      const wrapper = shallow(<PendingMessage pending={props} />)
      window.confirm = jest.fn(() => true)
      wrapper.find('button').simulate('click', { preventDefault() {}})
      expect(window.confirm).toBeCalled()
  })
});
