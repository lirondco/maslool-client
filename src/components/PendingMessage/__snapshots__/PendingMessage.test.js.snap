// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PendingMessage component renders the message given props 1`] = `
<div
  className="PendingMessage"
>
  <button
    aria-label="delete"
    onClick={[Function]}
  >
    x
  </button>
  <p
    className="outer_item"
  >
    From: 
    <span>
      test_user
    </span>
  </p>
  <p
    className="inner_item"
  >
    Content:
     
    <Link
      to="/messages/9"
    >
      <span>
        This is a message
         ... click to see more
      </span>
    </Link>
  </p>
  <p
    className="outer_item"
  >
    Contact:
     
    <span>
      <a
        href="mailto:test@user.no"
      >
        test@user.no
      </a>
    </span>
  </p>
</div>
`;
